polkadot-protocol/protocol-components/accounts/account-data-structure.md

```diff
@@ -0,0 +1,100 @@
+
+### Account Deactivation 
+
+Pallets also include cleanup functions that decrement the `providers` reference counter, marking the account as deactivated within the pallet-managed scope. 
+
+Once both the `providers` and `consumers` reference counters reach zero, the account is considered deactivated by all on-chain pallets.
+
+### Updating Counters
+
+Runtime developers can manage these counters using the:
+
+-  `inc_consumers()` - Increment the reference counter on an account.
+- `dec_consumers()` - Decrement the reference counter on an account.
+- `inc_providers()` - Increment the provider reference counter on an account.
+- `dec_providers()` - Decrement the provider reference counter on an account.
+- `inc_sufficients()` - Increment the self-sufficient reference counter on an account.
+- `dec_sufficients()` - Decrement the sufficients reference counter on an account.
+
+!!!note
+    All these methods are provided by the [`frame-system`](https://paritytech.github.io/polkadot-sdk/master/frame_system/pallet/struct.Pallet.html#method){target=\_blank} pallet.
+
+For every increment of a counter during an account's lifecycle, a corresponding decrement should be called to ensure proper management of the account state.
+
+There are also three query functions to ease usage on these counters:
+
+- [`can_inc_consumer()`](https://paritytech.github.io/polkadot-sdk/master/frame_system/pallet/struct.Pallet.html#method.can_inc_consumer){target=\_blank} to check if an account is ready to be used (`providers` > 0).
+- [`can_dec_provider()`](https://paritytech.github.io/polkadot-sdk/master/frame_system/pallet/struct.Pallet.html#method.can_dec_provider){target=\_blank} to check if an account is no longer referenced in runtime whatsoever (`consumers` == 0) before decrementing providers to 0.
+- [`is_provider_required()`](https://paritytech.github.io/polkadot-sdk/master/frame_system/pallet/struct.Pallet.html#method.is_provider_required){target=\_blank} to check if an account has outstanding consumer references (`consumers` > 0).
+
+!!! note
+    See [frame-system API](https://paritytech.github.io/polkadot-sdk/master/frame_system/pallet/struct.Pallet.html){target=\_blank} for more details.
+
+## AccountData Trait and Implementation
+
+The [`AccountInfo`](https://paritytech.github.io/polkadot-sdk/master/frame_system/struct.AccountInfo.html){target=\_blank} can be any structure, provided it meets the requirements of the associated type `AccountData` trait bound defined in the [`frame-system::pallet::Config`](https://paritytech.github.io/polkadot-sdk/master/frame_system/pallet/trait.Config.html){target=\_blank} trait. 
+
+By default, the Substrate runtime configures `AccountInfo` as defined in the [`pallet-balances`](https://paritytech.github.io/polkadot-sdk/master/pallet_balances/struct.AccountData.html){target=\_blank}.
```

Comment:
I've started to just say "runtime" instead "Substrate runtime" for consistency, @eshaben is this fine?

```suggestion
By default, the runtime configures `AccountInfo` as defined in the [`pallet-balances`](https://paritytech.github.io/polkadot-sdk/master/pallet_balances/struct.AccountData.html){target=\_blank}.
```